src/main/java/Frappe.java

```diff
@@ -0,0 +1,308 @@
+
+        String name = input[0].trim();
+        String doBy = input[1].trim();
+
+        if (name.isEmpty()) {
+            throw new FrappeException(FrappeException.NO_NAME);
+        }
+
+        Task task = new Deadline(name, doBy);
+        tasks.add(task);
+        Printer.printTaskAdded(task);
+    }
+
+    private static void addEvent(String[] words) throws FrappeException {
+        String[] preInput = Arrays.copyOfRange(words, 1, words.length);
+
+        if (preInput.length == 0) {
+            throw new FrappeException(FrappeException.NO_NAME);
+        }
+
+        String[] input = String.join(" ", preInput).split("/from");
+
+        if (input.length < 2) {
+            throw new FrappeException(FrappeException.NO_FROM);
+        }
+
+        if (input.length > 2) {
+            throw new FrappeException(FrappeException.TOO_MANY_FROM);
+        }
+
+        if (input[0].contains("/to")) {
+            throw new FrappeException(FrappeException.WRONG_TO_PLACEMENT);
+        }
+
+        String[] input2 = String.join(" ", input[1]).split("/to");
+
+        if (input2.length < 2) {
+            throw new FrappeException(FrappeException.NO_TO);
+        }
+
+        if (input2.length > 2) {
+            throw new FrappeException(FrappeException.TOO_MANY_TO);
+        }
+
+        String name = input[0].trim();
+        String from = input2[0].trim();
+        String to = input2[1].trim();
+
+        if (name.isEmpty()) {
+            throw new FrappeException(FrappeException.NO_NAME);
+        }
+
+        if (from.isEmpty()) {
+            throw new FrappeException(FrappeException.NO_FROM);
+        }
+
+        Task task = new Event(name, from, to);
+        tasks.add(task);
+        Printer.printTaskAdded(task);
+    }
+
+    private static void removeTask(String[] words) throws FrappeException {
+        int index = getTaskIndex(words);
+        Task task = tasks.get(index);
+        tasks.remove(task);
+        Printer.printTaskRemoved(task);
+    }
+
+    private static void processInput(String[] words) throws FrappeException {
+        switch (words[0]) {
+        case "list":
+            Printer.printTasks();
+            break;
+        case "mark":
+            updateTaskDone(words, true);
+            saveTasks();
+            break;
+        case "unmark":
+            updateTaskDone(words, false);
+            saveTasks();
+            break;
+        case "todo":
+            addTodo(words);
+            saveTasks();
+            break;
+        case "deadline":
+            addDeadline(words);
+            saveTasks();
+            break;
+        case "event":
+            addEvent(words);
+            saveTasks();
+            break;
+        case "delete":
+            removeTask(words);
+            saveTasks();
+            break;
+        default:
+            throw new FrappeException(FrappeException.UNKNOWN_COMMAND);
+        }
+    }
+
+    private static void saveTasks() throws FrappeFileException {
```

Comment:
I think you can SLAP harder here + add more abstraction

https://nus-cs2113-ay2425s2.github.io/website/se-book-adapted/chapters/codeQuality.html#slap-hard